health_vara.py

```diff
@@ -257,6 +310,47 @@ def get_most_recent_imaging_request_datetime(self, _name):
             dt = max(i.date for i in self.imaging_test_requests)
             return dt
 
+    @classmethod
+    def search_most_recent_imaging_request_datetime(cls, _name, clause):
+        name, op, value = clause
+
+        patient = cls.__table__()
+
+        ImagingTestRequest = Pool().get('gnuhealth.imaging.test.request')
+        imaging_test_requests = ImagingTestRequest.__table__()
+
+        patient_with_requests = patient.join(
+            imaging_test_requests,
+            'LEFT',
+            condition=patient.id == imaging_test_requests.patient
+        )
+
+        max_date_col = Max(imaging_test_requests.date).as_('max_date')
+        where = datetime_sql_where_clause(op, value, max_date_col)
+
+        grouped_patients = patient_with_requests.select(patient.id, max_date_col, group_by=patient.id)
+        max_date_query = grouped_patients.select(grouped_patients.id, where=where)
+        
+        return [('id', 'in', max_date_query)]
+
+    @classmethod
+    def order_most_recent_imaging_request_datetime(cls, tables):
```

Comment:
And ordering by the same field:
![image](https://github.com/vara-ai/health_vara/assets/1367955/81b1538b-d108-4c96-a404-6798ea55a11e)

Reply:
I have a feeling the "caching" here is actually never used, because it was broken in code, but it never broke while testing...
I have updated it to fix the code and match the style of
https://hg.tryton.org/modules/purchase_request_quotation/file/default/purchase.py#l334
but maybe I should just remove it.

Reply:
> I have a feeling the "caching" here is actually never used, because it was broken in code, but it never broke while testing...

What do you mean by caching in this context?

Reply:
I think I've read, and definitely seen, that the `tables` gets updated with new joins, and so at the start of a method call, there's often a check to see if the desired tables is already in `tables` (like in the link above, and line 344 here).

However, until my recent fix, the logic when the table already existed would not have worked, and should have blown up, since I was calling `max_date` on a `dict e.g. {}` rather than a table.

So my understanding is that `tables` should be `caching` the join that I add to it, but logging it out, it seems that's not true, and it always recalculates the  `grouped_requests` table.

Reply:
Since you matched the style of the `order_quotation_state` lookup and the logic is basically the same, is the caching for the `quotation` join in this method working?

Reply:
I'm not sure, that code I linked is an example from the official work, but not something we use ourselves. For now I will merge this, and I will try to investigate in parallel with other work as I feel this is low enough priority.